web3/net.py

```diff
@@ -31,14 +42,34 @@ def listening(self) -> bool:
     def peer_count(self) -> int:
         return self._peer_count()
 
-    @property
-    def peerCount(self) -> int:
-        return self._peerCount()
-
     @property
     def chainId(self) -> NoReturn:
         raise DeprecationWarning("This method has been deprecated in EIP 1474.")
 
     @property
     def version(self) -> str:
         return self._version()
+
+    #
+    # Deprecated Methods
+    #
+    peerCount = DeprecatedMethod(peer_count, 'peerCount', 'peer_count')  # type: ignore
+
+
+class AsyncNet(BaseNet):
+    is_async = True
+
+    @property
+    async def listening(self) -> bool:
+        # types ignored b/c mypy conflict with BlockingNet properties
```

Comment:
@cburgdorf I have a mypy question for you when you get a chance - We are having to ignore all of the types for methods in our Async modules, because the methods in the Blocking modules have the same name and signature. Other than ignoring the type or changing the names of the methods, do you know of a way around this? The idea was to be able to keep the same API for both async and sync, but if the cost of that is not having types for half the codebase, I'm not sure that's a good tradeoff.

Reply:
As far as I can tell you should be able to define `async def listening -> bool` here because the sync version isn't defined here or in the parent but in `Net` which isn't in our inheritance chain.

What it is complaining about is that you use `await self._listening()`. If you drop the `await` all of these should be fine and don't need the ignore. The reason is that you are calling the synchronous version so there isn't actually anything to `await` here.

```
    @property
    async def listening(self) -> bool:
        # types ignored b/c mypy conflict with BlockingNet properties
        return self._listening()

    @property
    async def peer_count(self) -> int:
        # types ignored b/c mypy conflict with BlockingNet properties
        return self._peer_count()

    @property
    async def version(self) -> str:
        # types ignored b/c mypy conflict with BlockingNet properties
        return self._version()
```